enhancements/installer/openstack-ipi-failure-domains.md

```diff
@@ -0,0 +1,254 @@
+      subnet: f584ce8c-bb54-422b-8e11-6a530be4c23e
+    - ...
+```
+There might be additional options in the domains, like flavors or BGP peers, but for now we will cover the minimum.
+
+Then they'll choose what Failure Domains to use when deploying the masters and workers directly in the machine pools:
+
+```yaml
+controlPlane:
+  name: master
+  platform:
+    openstack:
+      failureDomains:
+      - rack1
+      - rack2
+      - rack3
+  replicas: 3 // this will deploy 1 master in each domain
+compute:
+- name: worker
+  platform:
+    openstack:
+      ...
+      failureDomains:
+      - rack1
+      - rack2
+      - rack3
+      - rack4
+      - rack5
+      rootVolume:
+        size: 30
+        type: performance
+  replicas: 10 // this will deploy 2 workers per rack
+```
+
+The installer will create the control plane machines in a way that they can easily be adopted by the upcoming ControlPlane machineset.
+
+When deploying nodes on separate subnets, the user will need to update the `machineNetwork` option to list all the possible CIDRs on which the machines can be. The `machineNetwork` is the source for creating the security groups rules during installation. There's a [bug][bz-2095323] in the OpenStack implementation that only takes into account the first CIDR that we'll need to fix.
+
+However, once the bug will be fixed, the OpenShift administrators will have to provide a list of the CIDRs that will need access to the machine network:
+
+```yaml
+networking:
+  ...
+  machineNetwork:
+  - cidr: 192.168.25.0/24
+  - cidr: 192.168.26.0/24
+  - cidr: 192.168.27.0/24
+  - ...
+```
+
+For now, `machinesSubnet` is required to be set when deploying nodes on separate Failure Domains, and is then used to identify where to create the API and Ingress ports.
+
+Also, the installer will provide some validations in order to avoid deployment errors:
+
+- Having `failureDomains` in machinepool without `machinesSubnet` is an error.
+- machinepool `zones` for both compute and storage can't be used together with machinepool `failureDomains`.
+- Check that `subnets` in `failureDomains` actually exist (the same validation as machinesSubnet). Note that we can't easily verify if the subnet can actually be used for a given availability zone. This is up to the OpenShift administrators to figure out which subnet they can use for which domain.
+- Check that `failureDomain` subnets have a matching CIDR in `machineNetwork`.
+
+### Risks and Mitigations
+
+- Backward compatibility will have to be maintained and the "old" way to define `zones` to remain available.
+- UX will be reviewed by Field Engineers, partnering with a customer who needs this feature. Also our QE will
+  help to review it.
+
+### Drawbacks
+
+- The networking resources are pre-created and managed by the OpenStack administrators, and therefore not managed by the OpenShift cluster. Changes in the networking infrastructure will have to be planned and accordingly updated in the cluster.
```

Comment:
Since we potentially have two different admin teams that have to be involved in this (OSP admins and OCP admins), will we be able to validate the changes made by the OSP admins are correct?  Is it also possible to provide "hints" as to what changes the OSP admins have to do on behalf of the OCP admin.